analyses/NC_cd_2010/01_prep_NC_cd_2010.R

```diff
@@ -0,0 +1,83 @@
+###############################################################################
+# Download and prepare data for `NC_cd_2010` analysis
+# © ALARM Project, April 2022
+###############################################################################
+
+suppressMessages({
+    library(dplyr)
+    library(readr)
+    library(sf)
+    library(redist)
+    library(geomander)
+    library(cli)
+    library(here)
+    devtools::load_all() # load utilities
+})
+
+# Download necessary files for analysis -----
+cli_process_start("Downloading files for {.pkg NC_cd_2010}")
+
+path_data <- download_redistricting_file("NC", "data-raw/NC", year = 2010)
+
+# download the enacted plan.
+# 2010 plan from https://redistricting.lls.edu/state/north-carolina/
+path_enacted <- "data-raw/NC/NC_enacted/Rucho_Lewis_Congress_3.shp"
+
+cli_process_done()
+
+# Compile raw data into a final shapefile for analysis -----
+shp_path <- "data-out/NC_2010/shp_vtd.rds"
+perim_path <- "data-out/NC_2010/perim.rds"
+
+if (!file.exists(here(shp_path))) {
+    cli_process_start("Preparing {.strong NC} shapefile")
+    # read in redistricting data
+    nc_shp <- read_csv(here(path_data), col_types = cols(GEOID20 = "c")) %>%
+        join_vtd_shapefile(year = 2010) %>%
+        st_transform(EPSG$NC)  %>%
+        rename_with(function(x) gsub("[0-9.]", "", x), starts_with("GEOID"))
+
+    # add municipalities
+    d_muni <- make_from_baf("NC", "INCPLACE_CDP", "VTD")  %>%
+        mutate(GEOID = paste0(censable::match_fips("NC"), vtd)) %>%
+        select(-vtd)
+    d_cd <- make_from_baf("NC", "CD", "VTD")  %>%
+        transmute(GEOID = paste0(censable::match_fips("NC"), vtd),
+            cd_2010 = as.integer(cd))
```

Comment:
I think this should be called `cd_2010` as in `cd_2000 = as.integer(cd)`. A bit confusing, but the year after cd should be the year corresponding to the cycle. So the lines that was enacted in 2011 for the 2012 elections is called cd_2010. The lines that come from `make_from_baf` are lines before the redistricting, so they are 2000 cycle lines